master/main.go

```diff
@@ -1,5 +1,113 @@
 package main
 
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+
+	"./models"
+)
+
+var connections map[*websocket.Conn]bool
+var initializeDB = false
+var dropDB = false
+var session *mgo.Session
+var collUsers *mgo.Collection
+
+func authentiaction(loginAndPass []string) bool {
+	result := models.User{}
+	err := collUsers.Find(bson.M{"username": loginAndPass[0]}).One(&result)
+	if err != nil {
+		return false
+	} else if result.Password == loginAndPass[1] {
+		return true
+	} else {
+		return false
+	}
+}
+
+func authHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Header.Get("Origin") != "http://"+r.Host {
```

Comment:
This is unnecessary and actually wrong - you should receive a http.BadRequest (code 400) when you do an invalid http operation and it so happens that it's the default behavior of gorilla.websocket to send that response code when an invalid header is received.